test/tizen_capi/unittest_tizen_capi_lr_scheduler.cpp

```diff
@@ -0,0 +1,376 @@
+ */
+TEST(nntrainer_capi_lr_scheduler, create_destruct_05_n) {
+  ml_train_lr_scheduler_h handle;
+  int status;
+  status =
+    ml_train_lr_scheduler_create(&handle, ML_TRAIN_LR_SCHEDULER_TYPE_UNKNOWN);
+  EXPECT_EQ(status, ML_ERROR_INVALID_PARAMETER);
+}
+
+/**
+ * @brief Learning rate scheduler Create / Destruct Test (negative test)
+ */
+TEST(nntrainer_capi_lr_scheduler, create_destruct_06_n) {
+  ml_train_optimizer_h opt_handle;
+  ml_train_lr_scheduler_h lr_sched_handle;
+  int status;
+  status = ml_train_optimizer_create(&opt_handle, ML_TRAIN_OPTIMIZER_TYPE_SGD);
+  EXPECT_EQ(status, ML_ERROR_NONE);
+
+  status = ml_train_lr_scheduler_create(&lr_sched_handle,
+                                        ML_TRAIN_LR_SCHEDULER_TYPE_CONSTANT);
+  EXPECT_EQ(status, ML_ERROR_NONE);
+
+  status = ml_train_optimizer_set_lr_scheduler(opt_handle, lr_sched_handle);
+  EXPECT_EQ(status, ML_ERROR_NONE);
+
+  status = ml_train_lr_scheduler_destroy(lr_sched_handle);
+  EXPECT_EQ(status, ML_ERROR_INVALID_PARAMETER);
+
+  status = ml_train_optimizer_destroy(opt_handle);
+  EXPECT_EQ(status, ML_ERROR_NONE);
+}
+
+/**
+ * @brief Learning rate scheduler Create / Destruct Test (negative test)
+ */
+TEST(nntrainer_capi_lr_scheduler, create_destruct_07_n) {
+  int status = ML_ERROR_NONE;
+
+  ml_train_model_h model;
+  ml_train_optimizer_h optimizer;
+  ml_train_lr_scheduler_h lr_scheduler;
+
+  status = ml_train_model_construct(&model);
+  EXPECT_EQ(status, ML_ERROR_NONE);
+
+  status = ml_train_optimizer_create(&optimizer, ML_TRAIN_OPTIMIZER_TYPE_ADAM);
+  EXPECT_EQ(status, ML_ERROR_NONE);
+
+  status = ml_train_optimizer_set_property(optimizer, "beta1=0.002",
+                                           "beta2=0.001", "epsilon=1e-7", NULL);
+  EXPECT_EQ(status, ML_ERROR_NONE);
+
+  status = ml_train_lr_scheduler_create(&lr_scheduler,
+                                        ML_TRAIN_LR_SCHEDULER_TYPE_EXPONENTIAL);
+  EXPECT_EQ(status, ML_ERROR_NONE);
+
+  status = ml_train_lr_scheduler_set_property(
+    lr_scheduler, "learning_rate=0.0001", "decay_rate=0.96", "decay_steps=1000",
+    NULL);
+  EXPECT_EQ(status, ML_ERROR_NONE);
+
+  status = ml_train_model_set_optimizer(model, optimizer);
+  EXPECT_EQ(status, ML_ERROR_NONE);
+
+  status = ml_train_optimizer_set_lr_scheduler(optimizer, lr_scheduler);
+  EXPECT_EQ(status, ML_ERROR_NONE);
+
+  status = ml_train_lr_scheduler_destroy(lr_scheduler);
+  EXPECT_EQ(status, ML_ERROR_INVALID_PARAMETER);
+
+  status = ml_train_model_destroy(model);
+  EXPECT_EQ(status, ML_ERROR_NONE);
+}
+
+/**
+ * @brief Learning rate scheduler set Property Test (positive test)
+ */
+TEST(nntrainer_capi_lr_scheduler, setProperty_01_p) {
+  ml_train_lr_scheduler_h handle;
+  int status;
+  status =
+    ml_train_lr_scheduler_create(&handle, ML_TRAIN_LR_SCHEDULER_TYPE_CONSTANT);
+  EXPECT_EQ(status, ML_ERROR_NONE);
+  status =
+    ml_train_lr_scheduler_set_property(handle, "learning_rate=0.001", NULL);
```

Comment:
we need to check if the user sets this scheduler property and model property differently.

Reply:
We need to double check it but I guess in that case the optimizer will throw an error. (please refer https://github.com/nnstreamer/nntrainer/blob/main/nntrainer/optimizers/optimizer_wrapped.cpp#L96)